airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3ConsumerFactory.kt

```diff
@@ -138,8 +141,17 @@ class S3ConsumerFactory {
         storageOperations: BlobStorageOperations,
         writeConfigs: List<WriteConfig>
     ): OnCloseFunction {
-        return OnCloseFunction { hasFailed: Boolean, _: Map<StreamDescriptor, StreamSyncSummary> ->
+
+        val streamDescriptorToWriteConfig =
+            writeConfigs.associateBy {
+                StreamDescriptor().withNamespace(it.namespace).withName(it.streamName)
+            }
+        return OnCloseFunction {
+            hasFailed: Boolean,
+            streamSyncSummaries: Map<StreamDescriptor, StreamSyncSummary> ->
             if (hasFailed) {
+                // TODO: From the code looks like hasFailed is set to true after onClose is called
```

Comment:
fyi I think this call can set hasFailed=true during the sync https://github.com/airbytehq/airbyte/blob/1d24db9b29eacc92d6d20d986c7160d3e3a63d52/airbyte-cdk/java/airbyte-cdk/core/src/main/kotlin/io/airbyte/cdk/integrations/destination/async/AsyncStreamConsumer.kt#L99

(.... not that I fully understand what's happening here. Why do we want to delete the files we wrote, if there was an exception during the sync?)

Reply:
Oh interesting, missed that during accept it can fail too.  That makes 2 of us, not sure why old code was cleaning up all files on failure. We can make a call not to do it in this PR too considering this is a breaking change PR.

Reply:
Just confirming here: the correct behavior is just to return early on `hasFailed==true`?

Does that mean that in the case where a large overwrite sync fails halfway through we'll orphan the old data?

Scenario:
* Dest starts an overwrite, collects keys to delete
* Dest fails after checkpointing, deletes nothing
* Platform restarts at checkpoint
* Dest sees current gen Id in existing data and decides not to delete
* Dest succeeds
* "Keys to delete" from earlier run remain

Does it follow from this that `listExistingObjects` should actually take a genId filter and be called regardless?

Reply:
Specifically:

* Platform restarts at checkpoint
* Dest sees current gen id in existing data & recognizes we're in refresh
* Dest collects all keys with gen id < min gen id in metadata
* Dest succeeds
* Dest deletes those keys on success

Reply:
Completely removed the `hasFailed-> delete stored objects` logic